src/main/resources/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition/config.jelly

```diff
@@ -29,6 +29,9 @@ THE SOFTWARE.
     <f:entry field="scriptPath" title="${%Script Path}">
         <f:textbox default="Jenkinsfile"/>
     </f:entry>
+    <f:entry field="lightweight" title="${%Lightweight checkout}">
```

Comment:
I preferred to err on the side of leaving existing behavior alone by default for now.

* lack of changelog/polling support (see below)
* new code path with likely bugs compared to original